From 2b4049e1ecb47e41a4cc3eec58d22a02f14cfca5 Mon Sep 17 00:00:00 2001
From: Christian Hergert <chergert@redhat.com>
Date: Mon, 24 Jun 2024 12:53:35 -0700
Subject: [PATCH 2/2] vte.sh: forward-port notify/prexec/precmd

This brings forward the patches Fedora previously had to notify of the
command with precmd/preexec. They currently only affect bash.
---
 src/vte.sh.in | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/vte.sh.in b/src/vte.sh.in
index 2328a9ec..40cacd4f 100644
--- a/src/vte.sh.in
+++ b/src/vte.sh.in
@@ -32,11 +32,18 @@ __vte_osc7 () {
     printf "\033]7;file://%s%s\033\\" "${HOSTNAME}" "$(@libexecdir@/vte-urlencode-cwd)"
 }
 
+__vte_shell_precmd() {
+    local command=$(HISTTIMEFORMAT= history 1 | sed 's/^ *[0-9]\+ *//')
+    command="${command//;/ }"
+    printf '\033]777;notify;Command completed;%s\033\\\033]777;precmd\033\\' "${command}"
+}
+
 __vte_prompt_command() {
     local pwd='~'
     [ "$PWD" != "$HOME" ] && pwd=${PWD/#$HOME\//\~\/}
     pwd="${pwd//[[:cntrl:]]}"
     printf "\033]0;%s@%s:%s\033\\" "${USER}" "${HOSTNAME%%.*}" "${pwd}"
+    __vte_shell_precmd
     __vte_osc7
 }
 
@@ -49,9 +56,12 @@ if [[ -n "${BASH_VERSION:-}" ]]; then
     # use the __vte_prompt_command function which also sets the title.
 
     if [[ "$(declare -p PROMPT_COMMAND 2>&1)" =~ "declare -a" ]]; then
+	PROMPT_COMMAND+=(__vte_shell_precmd)
 	PROMPT_COMMAND+=(__vte_osc7)
+	PS0=$(printf "\033]777;preexec\033\\")
     else
 	PROMPT_COMMAND="__vte_prompt_command"
+	PS0=$(printf "\033]777;preexec\033\\")
     fi
 
     # Shell integration
-- 
2.45.1

